infra/gravity/ops_center.go

```diff
@@ -0,0 +1,159 @@
+package gravity
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	errClusterNotFound = errors.New("cluster not found")
+)
+
+// getTeleClusterStatus will attempt to get the cluster status from an ops center
+// for now, it's just looking for the specific line of output from tele get clusters
+// but could parse the entire yaml structure if imported from gravity
+func getTeleClusterStatus(clusterName string) (string, error) {
+	out, err := exec.Command("tele", "get", "clusters", clusterName, "--format", "yaml").Output()
+	if err != nil {
+		logrus.WithError(err).Error("unable to parse tele get clusters: ", string(out))
+		return "", trace.WrapWithMessage(err, string(out))
+	}
+
+	res, err := parseClusterStatus(clusterName, bytes.NewReader(out))
+	if err != nil {
+		logrus.WithError(err).Error("unable to parse tele get clusters: ", string(out))
+		return "", trace.WrapWithMessage(err, string(out))
+	}
+	return res, nil
+}
+
+// parseClusterStatus will look at the command output, and search for the string "status: abc" and return
+// the "abc" portion.
+func parseClusterStatus(clusterName string, rd io.Reader) (string, error) {
```

Comment:
why not use yaml or json parser? this logic seems error prone, I would replace it with proper parser.

Reply:
I was a little torn on this. The options I see are:

- Just try and parse the text
- duplicate the object structure into robotest, which would then open the possibility for it to diverge
- vendor gravity, and get the structs from the gravity source code

So when I looked at other sections of robotest, it looks like it was doing parsing, such as https://github.com/gravitational/robotest/blob/f9ff9f4c8a3ddaceacbc84c2438dbe26f8953364/infra/gravity/parse.go#L18

Anyway's, I'm not opposed to vendoring in gravity so I can use the existing object definitions if you think that's the best approach.

Reply:
I was thinking actually about unmarshalling the whole thing into substructure having only the fields you need, and it will be OK for now.